app/javascript/controllers/search_controller.js

```diff
@@ -1,36 +1,30 @@
 import { Controller } from '@hotwired/stimulus';
 
 export default class extends Controller {
-  static targets = ['loader', 'input'];
+  static targets = ['input'];
 
   connect() {
-    this.inputTarget.value = this.inputTarget.getAttribute('value');
+    this.moveCursorToEnd();
   }
 
-  disconnect() {
-    this.loaderTarget.classList.add('u-display-none');
-    this.inputTarget.value = '';
+  moveCursorToEnd() {
+    const searchValueLength = this.inputTarget.value.length;
+    if (searchValueLength > 0) {
+      this.inputTarget.focus();
+      this.inputTarget.setSelectionRange(searchValueLength, searchValueLength);
+    }
   }
 
-  query(event) {
-    const query = event.target.value.trim();
+  submit(e) {
```

Comment:
The submit function can write like this

```js
submit(e) {
    this.inputTarget.value = this.inputTarget.value.trim();

    if (this.inputTarget.value.length === 0) {
      e.preventDefault();
      e.stopPropagation();
    } else {
      this.element.classList.add('is-loading')
    }
  }
```
Now we can avoid use loading function to update status of loading.